lib/room.rb

```diff
@@ -0,0 +1,74 @@
+require_relative 'reservations'
+require 'date'
+
+module Hotel
+  class Room
+      attr_accessor :id, :reservations
+    def initialize(id:, reservations:)
+        @id = id
+        @reservations = []
+    end
+
+    def self.list_of_rooms
+      rooms = []
+      id = 1
+      20.times do
+          room = Hotel::Room.new(id: id, reservations: [])
+          rooms << room 
+          id += 1
+      end
+      return rooms
+    end
+
+    def list_of_reservations_for_data_range(start_date,end_date)
+      return reservations.select do |reservation|  
+          overlap1 = start_date >= reservation.date_in && start_date < reservation.date_out
+          overlap2 = end_date > reservation.date_in && end_date < reservation.date_out 
+          overlap3 = start_date >= reservation.date_in && end_date < reservation.date_out
+          overlap4 = start_date <= reservation.date_in && end_date >= reservation.date_out
+
+          overlap1 || overlap2 || overlap3 || overlap4
```

Comment:
There is a simpler way of doing this date checking! Make a timeline, put an existing reservation on the timeline, and then see where you can draw lines that don't conflict!